packages/core/src/actions/emit.ts

```diff
@@ -0,0 +1,152 @@
+import isDevelopment from '#is-development';
+import {
+  ActionArgs,
+  AnyActorScope,
+  AnyActor,
+  AnyMachineSnapshot,
+  DelayExpr,
+  EventObject,
+  MachineContext,
+  NoInfer,
+  RaiseActionOptions,
+  SendExpr,
+  ParameterizedObject,
+  AnyEventObject
+} from '../types.ts';
+
+function resolveEmit(
+  _: AnyActorScope,
+  snapshot: AnyMachineSnapshot,
+  args: ActionArgs<any, any, any>,
+  actionParams: ParameterizedObject['params'] | undefined,
+  {
+    event: eventOrExpr,
+    id,
+    delay
+  }: {
+    event:
+      | EventObject
+      | SendExpr<
+          MachineContext,
+          EventObject,
+          ParameterizedObject['params'] | undefined,
+          EventObject,
+          EventObject
+        >;
+    id: string | undefined;
+    delay:
+      | string
+      | number
+      | DelayExpr<
+          MachineContext,
+          EventObject,
+          ParameterizedObject['params'] | undefined,
+          EventObject
+        >
+      | undefined;
+  },
+  { internalQueue }: { internalQueue: AnyEventObject[] }
+) {
+  const delaysMap = snapshot.machine.implementations.delays;
+
+  if (typeof eventOrExpr === 'string') {
+    throw new Error(
+      `Only event objects may be used with emit; use emit({ type: "${eventOrExpr}" }) instead`
+    );
+  }
+  const resolvedEvent =
+    typeof eventOrExpr === 'function'
+      ? eventOrExpr(args, actionParams)
+      : eventOrExpr;
+
+  let resolvedDelay: number | undefined;
+  if (typeof delay === 'string') {
+    const configDelay = delaysMap && delaysMap[delay];
+    resolvedDelay =
+      typeof configDelay === 'function'
+        ? configDelay(args, actionParams)
+        : configDelay;
+  } else {
+    resolvedDelay =
+      typeof delay === 'function' ? delay(args, actionParams) : delay;
+  }
+  if (typeof resolvedDelay !== 'number') {
+    internalQueue.push(resolvedEvent);
+  }
```

Comment:
This is wrong because it would send that as a raised event to the current machine. It would be good to add a test case for this (to test that it doesn't happen)
```suggestion
```